src/Common/src/System/Security/Cryptography/Asn1/Pkcs12/PfxAsn.manual.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Security.Cryptography.Pkcs;
+
+namespace System.Security.Cryptography.Asn1.Pkcs12
+{
+    internal partial struct PfxAsn
+    {
+        internal bool VerifyMac(
+            ReadOnlySpan<char> macPassword,
+            ReadOnlySpan<byte> authSafeContents)
+        {
+            Debug.Assert(MacData.HasValue);
+
+            HashAlgorithmName hashAlgorithm;
+            int expectedOutputSize;
+
+            string algorithmValue = MacData.Value.Mac.DigestAlgorithm.Algorithm.Value;
+
+            switch (algorithmValue)
```

Comment:
we should create some utility for getting the size given HashAlgorithmName - I've seen this code too many times 😄

Reply:
There was already a proposal for getting the HashAlgorithmName from an OID, we can add overloads to that to also out the size (then get to debate if it's bytes or bits)